packages/card-support/src/components/DownloadClientData/utils.tsx

```diff
@@ -55,28 +59,53 @@ export const handleCardOrderDateChange = (
  * @param setSubmitting - method to set form `isSubmitting` status
  * @param langObj - tthe lang object
  */
-export const submitForm = (
+export const submitForm = async (
   values: DownloadClientDataFormFields,
   accessToken: string,
   opensrpBaseURL: string,
   serviceClass: typeof OpenSRPService,
   locations: ParsedHierarchyNode[],
   setSubmitting: (isSubmitting: boolean) => void,
   langObj: Lang = lang
-): void => {
+): Promise<void> => {
   const { clientLocation, cardStatus, cardOrderDate } = values;
 
   if (!clientLocation) {
     return;
   }
 
   setSubmitting(true);
-  const startDate = cardOrderDate[0];
-  const endDate = cardOrderDate[1];
-  const endPoint = `${OPENSRP_URL_CLIENT_SEARCH}`;
-  const serve = new serviceClass(accessToken, opensrpBaseURL, endPoint);
+
+  // get location id's of all nested children of a selected jurisdiction
+  const fetchNestedLocationIds = async (clientLocationId: string) => {
+    const serve = new serviceClass(accessToken, opensrpBaseURL, OPENSRP_URL_LOCATION_HIERARCHY);
+    try {
+      const res: RawOpenSRPHierarchy = await serve.read(clientLocationId);
+      // parentChildren is an object with keys as parent location ids
+      // and values as arrays of children location ids
+      // each children location with children itself has an entry in the object
+      const nestedLocationIds = Object.values(res.locationsHierarchy.parentChildren);
+      const flatNestedLocationIds = nestedLocationIds.flat();
+      // all keys (nested children location ids) are present as values of the primary location id values (clientLocationId)
+      // except the primary location id itself - thus the re-add
+      const flatNestedLocationIdsWithClientLocationId = [
+        ...flatNestedLocationIds,
+        clientLocationId,
+      ];
+      const stringifiedFlatNestedLocationIdsWithClientLocationId =
+        flatNestedLocationIdsWithClientLocationId.join(',');
+      return stringifiedFlatNestedLocationIdsWithClientLocationId;
+    } catch (_) {
+      sendErrorNotification(langObj.ERROR_OCCURRED);
+      return '';
+    }
+  };
+
+  const nestedLocationIds = await fetchNestedLocationIds(clientLocation);
+
+  // add all nested location ids (children of selected location) to the search params
   let params: Dictionary = {
-    locationIds: clientLocation,
+    locationIds: nestedLocationIds,
```

Comment:
Not blocking but I think worth to think about; 
would this potentially cause a 431 response, because the url can possibly be too long

Reply:
@dubdabasoduba could there be a better approach of generating the download data from the server? eg just passing the root location id and generating the nested locations from the server?

Reply:
@machariamuguku I think you have a great suggestion. We could create an issue for the backend team to create/update the API to work in this way. 

@p-netm I agree I haven't tested the endpoint with a large locations dataset.  This is something I can test and revert back.  

Do you guys mind merging this to unblock the prod blockage then we can add the enhancement after the API is updated?

Reply:
https://github.com/opensrp/opensrp-server-web/issues/1057